js/pro/timex.js

```diff
@@ -0,0 +1,622 @@
+'use strict';
+
+//  ---------------------------------------------------------------------------
+const timexRest = require ('../timex');
+const { ArgumentsRequired, ExchangeError } = require ('../base/errors');
+const { ArrayCache, ArrayCacheBySymbolById } = require ('./base/Cache');
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class timex extends timexRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchBalance': true,
+                'watchTicker': true,
+                'watchTickers': false,
+                'watchTrades': true,
+                'watchMyTrades': false,
+                'watchOrders': true,
+                'watchOrderBook': true,
+                'watchOHLCV': true,
+            },
+            'urls': {
+                'api': {
+                    'ws': 'wss://plasma-relay-backend.timex.io/socket/relay',
+                },
+            },
+            'options': {
+            },
+            'timeframes': {
+                '1m': 'I1',
+                '5m': 'I5',
+                '15m': 'I15',
+                '30m': 'I30',
+                '1h': 'H1',
+                '2h': 'H2',
+                '4h': 'H4',
+                '6h': 'H6',
+                '12h': 'H12',
+                '1d': 'D1',
+                '1w': 'W1',
+            },
+            'streaming': {
+            },
+            'exceptions': {
+            },
+        });
+    }
+
+    requestId () {
+        const requestId = this.sum (this.safeInteger (this.options, 'requestId', 0), 1);
+        this.options['requestId'] = requestId;
+        return requestId;
+    }
+
+    async watchBalance (params = {}) {
+        /**
+         * @method
+         * @name timex#watchBalance
+         * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @url https://docs.timex.io/websocket-api-general.html#subscribe-account-to-channels
+         * @param {object} params extra parameters specific to the timex api endpoint
+         * @returns {object} a [balance structure]{@link https://docs.ccxt.com/en/latest/manual.html?#balance-structure}
+         */
+        this.checkRequiredCredentials ();
+        const wallet = this.safeString2 (params, 'wallet', 'walletAddress', this.wallet);
```

Comment:
Am I looking at the wrong docs? I don't see the `account` option in there
https://docs.timex.io/websocket-api-list-streams.html#balances

Reply:
So the balances stream only returns desposits and withdrawals. So for our definition of `watchBalance` I subscribed to the account. (https://docs.timex.io/websocket-api-general.html#subscribe-account-to-channels)

```
2022-10-12T15:32:22.213Z onMessage {
  type: 'SUBSCRIBED',
  requestId: '1',
  subscriptionId: '2405fe59-ef35-4e73-9ae1-b61cdbd99a36',
  payload: { snapshot: { balances: [Array], openOrders: [Array] } }
}
{
  info: [
    { currency: 'SAI', totalBalance: '0', lockedBalance: '0' },
    {
      currency: 'USDT',
      totalBalance: '16.54435572',
      lockedBalance: '0'
    },
    {
      currency: 'TIME',
      totalBalance: '0.0404643972346651',
      lockedBalance: '0'
    }
    ...
  ],
  timestamp: undefined,
  datetime: undefined,
  SAI: { free: 0, used: 0, total: 0 },
  USDT: { free: 16.54435572, used: 0, total: 16.54435572 },
  TIME: { free: 0.0404643972346651, used: 0, total: 0.0404643972346651 },
  ...
  free: {
    SAI: 0,
  ....
  },
  used: {
    SAI: 0,
   ...
  },
  total: {
    SAI: 0,
    USDT: 16.54435572,
    TIME: 0.0404643972346651
...
  }
}
```